mpact/models.py

```diff
@@ -52,18 +52,16 @@ def __str__(self):
         return f"{self.id} - {self.title}"
 
     def save(self, *args, **kwargs):
-        schedule_changed = False
-        if self.pk:
-            try:
-                previous_model = GroupChat.objects.get(pk=self.pk)
-                schedule_changed = (
-                    self.schedule_start_date != previous_model.schedule_start_date or
-                    self.schedule_start_time != previous_model.schedule_start_time
-                )
-            except GroupChat.DoesNotExist:
-                # odd that we have a PK but nothing in the DB, but that's fine
-                # it's always safe to rebuild the schedule even if nothing changed
-                schedule_changed = True
+        try:
+            # self.pk is a Telegram ID
+            previous_model = GroupChat.objects.get(pk=self.pk)
+            schedule_changed = (
+                self.schedule_start_date != previous_model.schedule_start_date or
+                self.schedule_start_time != previous_model.schedule_start_time
+            )
+        except GroupChat.DoesNotExist:
+            # `previous_model` doesn't exist because this GroupChat is new
+            schedule_changed = False
```

Comment:
Are there unintended consequences to this?

Reply:
I think this implements the original intention of:
```python
        schedule_changed = False
        if self.pk:
            # ...
```
If the GroupChat doesn't exist, I don't think there will be a schedule for it.